### ISSUE TITLE
Expressions in action payloads

### ISSUE THREAD

**Issue by ralfhandl** [oasis-tcs/odata-specs#341]
## Use case

The history of department D15 (example data from [OData-Temporal]) shall be merged into D08. Assume we already have a function `toTimeslices` that converts `Departments('D15')/history` into the `TimesliceWithPeriod` format so that
```java
GET ~/Departments('D15')/history/Temporal.toTimeslices()
```
returns
```json
[{"Timeslice": {
   "From": "2010-01-01",
   "To": "2011-01-01",
   "Name": "Services",
   "Budget": 1100
}},
 {"Timeslice": {
   "From": "2011-01-01",
   "To": "9999-12-31",
   "Name": "Services",
   "Budget": 1700
}}]
```
If action parameters were specified in the URL, one could write:
```java
POST ~/Departments('D08')/history/Temporal.Update(deltaTimeslices=@delta)
&@delta=$root/Departments('D15')/history/Temporal.toTimeslices()
```
But action parameters must be passed in the payload, and this does not allow aliases and expressions.

## Suggestion
```java
POST ~/Departments('D08')/history/Temporal.Update

{
 "deltaTimeslices@expressionUrl":
  "$root/Departments('D15')/history/Temporal.toTimeslices()"
}
```
Or, because in a structured payload, a structured expression fits better than an expression string that must be parsed, we borrow notation from [CSDL](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#sec_ApplyClientSideFunctions):
```java
POST ~/Departments('D08')/history/Temporal.Update

{
 "deltaTimeslices@expressionCsdl": {
  "$Function": "Temporal.toTimeslices",
  "$Apply": [
   {"$Path": "$root/Departments('D15')/history"}
  ]
 }
}
```

Imported from [ODATA-1505](https://issues.oasis-open.org/browse/ODATA-1505)

**Comment by HeikoTheissen** [comment]
Close only after #1943 is merged.
